### DISCUSSION TITLE
Changes not reflected in HA Media Player

### DISCUSSION THREAD

**Discussion by jamesonuk** [alandtse/alexa_media_player#2320] [Q&A]
Not sure if I have missed something but I am struggling to work out if I have done something wrong or whether this is a limitation of how Alexa works.

The main thing at the moment is that I want to run an automation that ups the volume, makes an announcement and then returns the volume to the previous level.   I have all this setup using a scene to snapshot the media player and it kind of works.   The issue is however the media player is not updating...    eg. I have set the volume to 10%  but HA is still showing the previous 40% value (I have given it time, tried setting volume in the app / through voice control but nothing is updated in HA).

Lots of things have been played on this device but I see that whilst it is updating, it isn't changing
![image](https://github.com/user-attachments/assets/09a51139-1cb3-40fc-a4d2-8be372d6a673)

The media player also has a state of standby.   I have turned it on in HA and now it shows the media that was playing from Spotify in HA and has now update the volume but if I change the volume again it is sticking at 10%.

If I toggle the media player in HA into / out of standby it picks up the volume but this seems a bit overkill.

Is this a latency issue, something I haven't done right or is there a better way I can capture the current volume ?

**Comment by zraken** [comment]
I don't think anything is updating anymore reliably. I heavily relied on last_called attribute but this updates at random after some seemingly long and random delay.

**Reply by danielbrunt57** [reply]
Yes, Amazon has ceased issuing the PUSHACTIVITY command via HTTP2 thus rendering last_alexa only able to update when a periodic poll occurs (auto = 10 minutes if scan interval is 1 minute and HTTP2 push is active, or you manually update last_alexa via the service call) and I see no other way...
